connectivity/check/wait.go

```diff
@@ -134,6 +135,29 @@ func WaitForCoreDNS(ctx context.Context, log Logger, client Pod) error {
 	}
 }
 
+// Service waits until the specified service is created and can be retrieved.
+func WaitForServiceRetrieval(ctx context.Context, log Logger, client *k8s.Client, namespace string, name string) (Service, error) {
+	log.Logf("⌛ [%s] Retrieving service %s/%s ...", client.ClusterName(), namespace, name)
+
+	ctx, cancel := context.WithTimeout(ctx, ShortTimeout)
+	defer cancel()
+	for {
+		svc, err := client.GetService(ctx, namespace, name, metav1.GetOptions{})
+		if err == nil {
+			return Service{Service: svc.DeepCopy()}, nil
+		}
+
+		log.Debugf("[%s] Failed to retrieve Service %s/%s: %s", client.ClusterName(), namespace, name, err)
+
+		select {
+		case <-time.After(PollInterval):
```

Comment:
Might be better to use inctimer here

Reply:
Hmm, I'd personally keep `time.After` for consistency with all other usages (`inctimer` doesn't seem to be used at all in Cilium CLI), and a global change would be out of scope here. It shouldn't matter much in terms of performance, as the CLI is not a long-running application as the Cilium agent, and it is far from being optimized in terms of memory consumption.